activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/ServiceTaskIntegrationResultEventHandler.java

```diff
@@ -52,9 +52,12 @@ public synchronized void receive(IntegrationResultEvent integrationResultEvent)
             runtimeService.trigger(integrationContext.getExecutionId(),
                                    integrationResultEvent.getVariables());
         } else {
-            LOGGER.warn("No task is waiting for integration result with execution id `" +
-                                integrationResultEvent.getExecutionId() +
-                                "`. The integration result `" + integrationResultEvent.getId() + "` will be ignored." );
+            String message = "No task is waiting for integration result with execution id `" +
+                    integrationResultEvent.getExecutionId() +
+                    "`. The integration result `" + integrationResultEvent.getId() + "` will be ignored.";
+            LOGGER.error( message );
+            // This needs to throw an exception so the message goes back to the queue in case of other node can pick it up.
+            throw new IllegalStateException(message);
```

Comment:
We need add a test covering this path.

Reply:
This is a tough test, because it involves testing RabbitMQ retry policy, but I totally agree, we need that test.

Reply:
Well, here was more thinking about an unit tests, but it might be interesting to have this integration test as well.